pkgs/native_assets_cli/test/api/build_config_test.dart

```diff
@@ -0,0 +1,218 @@
+
+    final fromConfig = BuildConfig.fromConfig(config);
+    expect(fromConfig, equals(buildConfig2));
+  });
+
+  test('BuildConfig.dryRun', () {
+    final buildConfig2 = BuildConfig.dryRun(
+      outDir: outDirUri,
+      packageName: packageName,
+      packageRoot: packageRootUri,
+      targetOs: OS.android,
+      linkModePreference: LinkModePreference.preferStatic,
+    );
+
+    final config = Config(fileParsed: {
+      'dry_run': true,
+      'link_mode_preference': 'prefer-static',
+      'out_dir': outDirUri.toFilePath(),
+      'package_name': packageName,
+      'package_root': packageRootUri.toFilePath(),
+      'target_os': 'android',
+      'version': BuildOutput.version.toString(),
+    });
+
+    final fromConfig = BuildConfig.fromConfig(config);
+    expect(fromConfig, equals(buildConfig2));
+  });
+
+  test('BuildConfig == dependency metadata', () {
+    final buildConfig1 = BuildConfig(
+      outDir: outDirUri,
+      packageName: packageName,
+      packageRoot: tempUri,
+      targetArchitecture: Architecture.arm64,
+      targetOs: OS.android,
+      targetAndroidNdkApi: 30,
+      buildMode: BuildMode.release,
+      linkModePreference: LinkModePreference.preferStatic,
+      dependencyMetadata: {
+        'bar': const Metadata({
+          'key': 'value',
+          'foo': ['asdf', 'fdsa'],
+        }),
+        'foo': const Metadata({
+          'key': 321,
+        }),
+      },
+    );
+
+    final buildConfig2 = BuildConfig(
+      outDir: outDirUri,
+      packageName: packageName,
+      packageRoot: tempUri,
+      targetArchitecture: Architecture.arm64,
+      targetOs: OS.android,
+      targetAndroidNdkApi: 30,
+      buildMode: BuildMode.release,
+      linkModePreference: LinkModePreference.preferStatic,
+      dependencyMetadata: {
+        'bar': const Metadata({
+          'key': 'value',
+        }),
+        'foo': const Metadata({
+          'key': 123,
+        }),
+      },
+    );
+
+    expect(buildConfig1, equals(buildConfig1));
+    expect(buildConfig1 == buildConfig2, false);
+    expect(buildConfig1.hashCode == buildConfig2.hashCode, false);
+  });
+
+  test('BuildConfig fromArgs', () async {
+    final buildConfig = BuildConfig(
+      outDir: outDirUri,
+      packageName: packageName,
+      packageRoot: tempUri,
+      targetArchitecture: Architecture.arm64,
+      targetOs: OS.android,
+      targetAndroidNdkApi: 30,
+      buildMode: BuildMode.release,
+      linkModePreference: LinkModePreference.preferStatic,
+    );
+    final configFileContents =
+        (buildConfig as internal.BuildConfig).toYamlString();
+    final configUri = tempUri.resolve('config.yaml');
+    final configFile = File.fromUri(configUri);
+    await configFile.writeAsString(configFileContents);
+    final buildConfig2 = await BuildConfig.fromArgs(
+      ['--config', configUri.toFilePath()],
+      environment: {}, // Don't inherit the test environment.
+    );
+    expect(buildConfig2, buildConfig);
```

Comment:
```suggestion
    expect(buildConfigFromArgs, buildConfig);
```
Maybe go over the variable names in this and other files, just to check if they are descriptive enough.